### PR TITLE
Fix bug in function body verifier

### PR DESCRIPTION
**Pull request by gramalingam**
The partial-checker for function-body has a bug where it uses the formal-parameter's name instead of actual-parameter's name. This PR fixes this bug.

### CODE REVIEW

File: onnx/test/cpp/function_verify_test.cc [hunk 1]
```diff
@@ -13,33 +13,29 @@ using namespace checker;
 using TENSOR_TYPES_MAP =
     std::unordered_map<std::string, std::vector<std::string>>;
 
+// Unfortunately, the different data-structures use different int types to
+// represent a parameter-index.
+using parameter_index = unsigned int;
```
Comment (ebarsoum) [inline onnx/test/cpp/function_verify_test.cc]: Why not use size_t and cast based on that?

File: onnx/test/cpp/function_verify_test.cc [hunk 2]
```diff
@@ -49,51 +45,53 @@ void VerifyTypeConstraint(
...
-        if (allowed_types.find(type) == allowed_types.end()) {
-          fail_check(
-              "Input type " + type + " defined in " + schema->Name() +
-              "'s function body is not allowed in node " + op_type);
+    // Check inputs of node:
+    parameter_index num_formal_inputs =
+        static_cast<parameter_index>(schema->inputs().size());
+    parameter_index num_actual_inputs = node.input_size();
+    // The above may not be same due to optional inputs
+    auto num_checked_inputs = std::min(num_formal_inputs, num_actual_inputs);
```
Comment (ebarsoum) [inline onnx/test/cpp/function_verify_test.cc]: Is there a case where actual inputs > formal input?

Comment (gramalingam) [reply onnx/test/cpp/function_verify_test.cc]: No. I can add a check and throw an error in this case, if that's what you mean.

Comment (ebarsoum) [reply onnx/test/cpp/function_verify_test.cc]: What I mean why not just use actual inputs?

Comment (gramalingam) [reply onnx/test/cpp/function_verify_test.cc]: Actually: sorry, I was wrong. This is possible with variadic parameters. I will modify the logic to do the right thing.

**Pull request merged**